src/main/scala/za/co/absa/abris/avro/registry/AbrisRegistryClient.scala

```diff
@@ -16,14 +16,21 @@
 
 package za.co.absa.abris.avro.registry
 
-import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata
+import org.apache.avro.Schema
 
-/**
- * This custom registry client can be used as a super class for custom registry clients that differ from the standard
- * Confluent [[io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient]].
- * These must implement a constructor that takes a map as input that is the one configured in the AbrisConfig.
- * The class is constructed in the Spark driver and the Spark executor.
- */
-abstract class CustomRegistryClient(config: java.util.Map[String, String]) extends SchemaRegistryClient {
+trait AbrisRegistryClient {
```

Comment:
What is the reason for this refactoring of the SchemaRegistryClient?

Reply:
To remove the dependency on Confluent code, now it will be encapsulated in one client implementation. If there are more changes in the future, we can deal about it inside the client without influencing the rest of the project.

Reply:
One such case was `MyRegistry.scala` that would need to differ in different confluent versions.
That class was removed.